src/DynamoCore/Models/DynamoModelCommands.cs

```diff
@@ -83,43 +83,55 @@ private void CreateAndConnectNodeImpl(CreateAndConnectNodeCommand command)
         {
             using (CurrentWorkspace.UndoRecorder.BeginActionGroup())
             {
-                var newNode = CreateNodeFromNameOrType(command.ModelGuid, command.NewNodeName);
-                newNode.X = command.X;
-                newNode.Y = command.Y;
-                var existingNode = CurrentWorkspace.GetModelInternal(command.ModelGuids.ElementAt(1)) as NodeModel;
-                
-                if(newNode == null || existingNode == null) return;
-
-                AddNodeToCurrentWorkspace(newNode, false, command.AddNewNodeToSelection);
-                CurrentWorkspace.UndoRecorder.RecordCreationForUndo(newNode);
-
-                PortModel inPortModel, outPortModel;
-                if (command.CreateAsDownstreamNode)
-                {
-                    // Connect output port of Existing Node to input port of New node
-                    outPortModel = existingNode.OutPorts[command.OutputPortIndex];
-                    inPortModel = newNode.InPorts[command.InputPortIndex];
-                }
-                else
-                {
-                    // Connect output port of New Node to input port of existing node
-                    outPortModel = newNode.OutPorts[command.OutputPortIndex];
-                    inPortModel = existingNode.InPorts[command.InputPortIndex];
-                }
+                CreateAndConnectNodeImplWithUndoGroup(command);
+            }
+        }
+
+        /// <summary>
+        /// This method assumes that there exists an undo-redo action group already
+        /// that can be used to record creation and deletion of models.
+        /// </summary>
+        /// <param name="command"></param>
+        private void CreateAndConnectNodeImplWithUndoGroup(CreateAndConnectNodeCommand command)
+        {
+            var newNode = CreateNodeFromNameOrType(command.ModelGuid, command.NewNodeName);
 
-                var models = GetConnectorsToAddAndDelete(inPortModel, outPortModel);
+            newNode.X = command.X;
```

Comment:
I guess from command standpoint, newNode could be null when the node name is invalid? So it could fail or crash? This is not something this PR brings but would like to confirm what would be the behavior.

Reply:
Yes, you're right, `newNode` can be null. I will add a null check.